### PR TITLE
nbstripout: fix tests

### PR DESCRIPTION
**Pull request by jluttine**
###### Motivation for this change

Building of nbstripout has failed after the version was bumped to 0.3.1 because:
the test files were missing from the new release tarball. This adds tests by
using the release tarball from GitHub instead of PyPI.

However, these tests fail because of a bug in one test. Thus, a patch is used to
fix the tests.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jluttine** [comment]
Note: The changes only affect the tests, that is, the check phase. The actual application remains untouched. These changes just fix the build of the derivation.

File: pkgs/applications/version-management/nbstripout/default.nix [hunk 1]
```diff
@@ -12,11 +12,30 @@ buildPythonApplication rec {
...
 
+  patches = [
+    (
+      # Fix git diff tests by using --no-index.
+      # See: https://github.com/kynan/nbstripout/issues/74
+      #
+      # Remove this patch once the pull request has been merged and a new
+      # release made.
+      fetchurl {
+        url = "https://github.com/kynan/nbstripout/compare/1056b101a0ecf40d7c8722f95632f8c37c9c7fb4...03e28424fb788dd09a95e99814977b0d0846c0b4.patch";
```
[the remark this answers is not in the document]
Comment (jluttine) [reply pkgs/applications/version-management/nbstripout/default.nix]: Done

**Comment by Mic92** [comment]
@GrahamcOfBorg build nbstripout

**Comment by jluttine** [comment]
Hmm.. Which branch does that mean? If it builds ok, no need to backport. If it fails and this fixes, then I'd backport. But what is the branch it would be backported to? nixos-18.03?

**Comment by jluttine** [comment]
It seems that on 18.03, nbstripout is still an older version which builds fine. So there is no need to backport this if the upgrade from 0.3.0 to 0.3.1 isn't backported either.

**Pull request merged**